docker/Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM ubuntu:16.04
+
+# Update
+RUN apt-get update && apt-get dist-upgrade -y
+
+# Install the necessary packages.
+RUN apt-get install -y \
+    supervisor apache2 git php7.0 libapache2-mod-php7.0 php7.0-cli \
+    php7.0-mysql php7.0-xml php7.0-mbstring php7.0-curl php7.0-zip
+
+RUN a2enmod php7.0
+RUN a2enmod rewrite
+
+# Add the files to the container.
+RUN mkdir /var/www/polr
+COPY . /var/www/polr/
+
+# Remove the storage directories, our startup script and symlinking will handle this.
+RUN rm -rf /var/www/polr/storage
+
+# Put the apache configuration in place
+RUN mv /var/www/polr/docker/apache.conf /etc/apache2/sites-available/000-default.conf
+
+# Configure supervisor to keep apache running.
+RUN mv /var/www/polr/docker/supervisor.conf /etc/supervisor/conf.d/apache.conf
+
+# Install and run composer to add necessary packages. Personally, I would prefer it if vendor was
+# bundled with releases, and not have this section
+RUN apt-get install curl  -y
+RUN curl -sS https://getcomposer.org/installer | php
+RUN mv composer.phar /usr/bin/composer
+RUN chmod +x /usr/bin/composer
+WORKDIR /var/www/polr
+RUN composer install
```

Comment:
Yes, I didn't like this at the time either, but had decided it was the best option available to me. In my own projects, I make sure to keep use `.gitignore` to keep vendor out of the dev branches as well, but I leave it in with the official releases to solve this problem. However I understand that this manual process is non-automated and tedious, and could have potential licensing issues (are all the packages "free" to distribute). I figure that it is normal to expect  developers to perform a composer update/install when using master or branches, but "average joe" who is not a developer who just want to deploy a release of this tool to a [raw docker server](https://www.digitalocean.com/products/one-click-apps/docker/) that doesn't have anything installed, shouldn't be forced to perform the extra steps of installing and using composer to pull additional files. I figured that since this is a virtual container environment, running composer as root inside it was "acceptable". This will not result in those scripts having access to anything on the host's computer. 

There is the additional choice of having the `build.sh` script install composer and perform the composer install comand and then perform composer install before kicking off the docker build.

Let me know your thoughts and I'll be happy to make the change.